plugins/qtosgrave/osgviewerwidget.cpp

```diff
@@ -32,6 +33,149 @@
 
 namespace qtosgrave {
 
+class OpenRAVETracker : public osgGA::NodeTrackerManipulator {
+public:
+    OpenRAVETracker()
+    {
+        _currentTransitionAnimationTime = 0;
+        _transitionAnimationPath = new osg::AnimationPath();
+        _transitionAnimationPath->setLoopMode(osg::AnimationPath::NO_LOOPING);
+    }
+
+    void TrackNode(osg::Node* node, const osg::Vec3d& offset, double trackDistance, osg::Camera* currentCamera, const osg::Vec3d& worldUpVector)
+    {
+        _offset = offset;
+        _distance = trackDistance;
+        osg::NodePathList nodeParents = node->getParentalNodePaths();
+        if(nodeParents.empty()) {
+            RAVELOG_WARN("Could not track node, node has no transform chain");
+            return;
+        }
+        NodeTrackerManipulator::setTrackNodePath(nodeParents[0]);
+        _transitionAnimationDuration = 1.0; //< transition animation time
+        _currentTransitionAnimationTime = 0;
+        _CreateTransitionAnimationPath(node, currentCamera, worldUpVector);
+        _time.restart();
+    }
+
+    // OSG overloaded methods
+public:
+
+    virtual osg::Matrixd getMatrix() const
+    {
+        osg::Vec3d nodeCenter;
+        osg::Quat nodeRotation;
+        computeNodeCenterAndRotation(nodeCenter, nodeRotation);
+
+        osg::Matrixd localToWorld;
+        computeNodeLocalToWorld(localToWorld);
+
+        nodeCenter = osg::Vec3d(_offset) * localToWorld;
+
+        return osg::Matrixd::translate(0.0,0.0,_distance) * osg::Matrixd::rotate(_rotation) * osg::Matrix::translate(nodeCenter);
+    }
+
+    // need to reimplement this method so we can track based on nodes origin instead of center of bounding sphere
+    virtual osg::Matrixd getInverseMatrix() const
+    {
+        if(!_IsTransitionAnimationFinished()) {
+            // a very small const encapsulation break, since this allow all animation logic to be totally contained in the manipulator
+            return const_cast<OpenRAVETracker*>(this)->_GetTransitionAnimationMatrix();
+        }
+
+        osg::Vec3d nodeCenter;
+        osg::Quat nodeRotation;
+        computeNodeCenterAndRotation(nodeCenter,nodeRotation);
+
+        osg::Matrixd localToWorld, worldToLocal;
+        computeNodeLocalToWorld(localToWorld);
+
+        nodeCenter = osg::Vec3d(_offset) * localToWorld;
+        return osg::Matrixd::translate(-nodeCenter) * osg::Matrixd::rotate(_rotation.inverse()) * osg::Matrixd::translate(0.0,0.0,-_distance);
+    }
+
+    virtual void setByInverseMatrix(const osg::Matrixd& matrix)
+    {
+        _rotation = matrix.getRotate().inverse();
+    }
+
+private:
+    void _CreateTransitionAnimationPath(osg::Node* node, osg::Camera* currentCamera, const osg::Vec3d& worldUpVector)
```

Comment:
@pablocael if there's no reason to use raw pointer, let's not. You can do osg::Node& node. thanks

Reply:
node not used here. Removed.